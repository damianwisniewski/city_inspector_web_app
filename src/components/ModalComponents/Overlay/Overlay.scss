.Modal-overlay {
	position: fixed;
	top: 0;
	left: 0;
	z-index: 1000;

	display: flex;
	align-items: center;
	justify-content: center;
	width: 100vw;
	height: 100%;

	&--dark {
		background-color: rgba(0, 0, 0, 0.6);

		animation: 0.4s 1 forwards darken;
	}

	&--transparent {
		background-color: transparent;
	}
}

@keyframes darken {
	from {
		background-color: rgba(0, 0, 0, 0);
	}

	to {
		background-color: rgba(0, 0, 0, 0.6);
	}
}
