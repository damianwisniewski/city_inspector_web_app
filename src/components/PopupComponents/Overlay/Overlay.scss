.popup-overlay {
	display: flex;
	justify-content: center;
	align-items: center;
	position: fixed;
	top: 0;
	left: 0;
	z-index: 500;
	width: 100vw;
	height: 100vh;

	&.dark {
		background-color: rgba(0, 0, 0, 0.5);
		animation-delay: 0.2s;
		animation: 0.5s 1 forwards darken;
	}

	&.transparent {
		background-color: transparent;
	}
}

@keyframes darken {
	from {
		background-color: rgba(0, 0, 0, 0);
	}

	to {
		background-color: rgba(0, 0, 0, 0.4);
	}
}
