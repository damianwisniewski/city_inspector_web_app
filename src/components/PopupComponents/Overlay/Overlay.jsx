import './Overlay.scss'
import React from 'react'
import PropTypes from 'prop-types'

const Overlay = ({ dark, onOverlayClick, children }) => {
	const handleClick = e => {
		if (e.target.id === 'overlay') {
			onOverlayClick()
		}
	}

	return (
		<div
			id='overlay'
			onClick={handleClick}
			className={`popup-overlay ${dark ? 'popup-overlay--dark' : 'popup-overlay--transparent'}`}
		>
			{children}
		</div>
	)
}

Overlay.defaultProps = {
	dark: false,
	onOverlayClick: () => {},
}

Overlay.propTypes = {
	dark: PropTypes.bool,
	onOverlayClick: PropTypes.func,
	children: PropTypes.element.isRequired,
}

export default Overlay
