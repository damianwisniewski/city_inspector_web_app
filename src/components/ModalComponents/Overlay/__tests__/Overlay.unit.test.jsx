import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Overlay from '../Overlay'

configure({ adapter: new Adapter() })

describe('(Modal) <Overlay />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let OverlayComponent
	const dark = true
	const children = <div />
	const onOverlayClick = jest.fn()

	beforeEach(() => {
		OverlayComponent = shallow(
			<Overlay dark={dark} onOverlayClick={onOverlayClick}>
				{children}
			</Overlay>,
		)
	})

	test('should render correctly Overlay with proper classNames, for dark prop equal true', () => {
		OverlayComponent.setProps({ dark: true })
		expect(toJson(OverlayComponent)).toMatchSnapshot()
	})

	test('should render correctly Overlay with proper classNames, for dark prop equal false', () => {
		OverlayComponent.setProps({ dark: false })
		expect(toJson(OverlayComponent)).toMatchSnapshot()
	})

	test('after click, should call function passed as onOverlayClick prop', () => {
		OverlayComponent.simulate('click', {
			target: {
				id: OverlayComponent.props().id,
			},
		})

		expect(onOverlayClick).toBeCalledTimes(1)
	})
})
