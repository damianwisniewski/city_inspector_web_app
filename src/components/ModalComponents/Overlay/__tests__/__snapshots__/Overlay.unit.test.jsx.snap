// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Modal) <Overlay /> should render correctly Overlay with proper classNames, for dark prop equal false 1`] = `
<div
  className="Modal-overlay Modal-overlay--transparent"
  id="overlay"
  onClick={[Function]}
>
  <div />
</div>
`;

exports[`(Modal) <Overlay /> should render correctly Overlay with proper classNames, for dark prop equal true 1`] = `
<div
  className="Modal-overlay Modal-overlay--dark"
  id="overlay"
  onClick={[Function]}
>
  <div />
</div>
`;
